reclass/datatypes/tests/test_applications.py

```diff
@@ -63,7 +63,7 @@ def test_repr_negations_interspersed(self):
         l = ['a', '~b', 'a', '~d']
         a = Applications(l)
         is_negation = lambda x: x.startswith(a.negation_prefix)
-        GOAL = filter(lambda x: not is_negation(x), set(l)) + filter(is_negation, l)
+        GOAL = list(filter(lambda x: not is_negation(x), set(l))) + list(filter(is_negation, l))
```

Comment:
This is probably the least efficient code to do this. itertools, past, future, or builtins might be better ways to get at this.